### ISSUE TITLE
Agnular-cli --aot build result in error on modal feature in 0.8.0

### ISSUE THREAD

**Issue by tarjelavik** [edcarroll/ng2-semantic-ui#88]
Thanks for the angularization of Semantic-UI! 

I just updated to `0.8.0` and tried to run `ng build --aot` and got errors on the new modal. Downgrading to `0.7.0` and the build was successful.

```bash
ERROR in /.../node_modules/ng2-semantic-ui/components/modal/modal.service.d.ts (9,26): Generic type 'ModalConfig<T, U>' requires 2 type argument(s).

ERROR in /.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (8,39): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (8,41): Type parameter declaration expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (9,24): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (10,19): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (11,15): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (12,20): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (13,26): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (14,21): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (15,23): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (16,31): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (19,47): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (19,49): Type parameter declaration expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (19,65): Expression expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (20,37): '(' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (23,48): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (23,50): Type parameter declaration expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (23,66): Expression expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (24,24): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (8,41): Left side of comma operator is unused and has no side effects.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (8,47): Cannot find name 'V'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (8,51): Object is possibly 'null'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (9,17): Cannot find name 'boolean'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (10,18): Cannot find name 'V'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (11,14): Cannot find name 'T'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (13,19): Cannot find name 'boolean'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (14,14): Cannot find name 'boolean'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (15,17): Cannot find name 'string'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (16,25): Cannot find name 'number'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (17,27): Cannot find name 'T'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (19,49): Left side of comma operator is unused and has no side effects.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (19,55): Cannot find name 'V'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (19,59): Object is possibly 'null'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (19,73): Left side of comma operator is unused and has no side effects.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (19,73): Left side of comma operator is unused and has no side effects.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (19,85): Cannot find name 'T'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (19,88): Cannot find name 'U'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (19,91): Cannot find name 'V'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (20,15): 'ModalTemplate' only refers to a type, but is being used as a value here.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (21,41): Cannot find name 'T'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (21,44): Cannot find name 'U'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (21,47): Cannot find name 'V'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (21,61): Cannot find name 'T'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (23,50): Left side of comma operator is unused and has no side effects.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (23,56): Cannot find name 'V'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (23,60): Object is possibly 'null'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (23,74): Left side of comma operator is unused and has no side effects.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (23,74): Left side of comma operator is unused and has no side effects.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (23,86): Cannot find name 'T'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (23,89): Cannot find name 'U'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (23,92): Cannot find name 'V'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-config.d.ts (25,48): Cannot find name 'T'.

ERROR in /.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (7,33): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (7,35): Type parameter declaration expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (7,51): Expression expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (8,15): ',' expected.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (7,35): Left side of comma operator is unused and has no side effects.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (7,41): Cannot find name 'V'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (7,45): Object is possibly 'null'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (7,59): Left side of comma operator is unused and has no side effects.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (7,73): Cannot find name 'U'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (7,76): Cannot find name 'V'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (8,14): Cannot find name 'T'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (9,41): Cannot find name 'U'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (9,44): Cannot find name 'V'.
/.../node_modules/ng2-semantic-ui/components/modal/modal-controls.d.ts (9,57): Cannot find name 'T'.

ERROR in /.../node_modules/ng2-semantic-ui/components/modal/active-modal.d.ts (8,27): Generic type 'ModalConfig<T, U>' requires 2 type argument(s).

ERROR in /.../node_modules/ng2-semantic-ui/components/modal/modal.d.ts (29,27): Generic type 'ModalConfig<T, U>' requires 2 type argument(s).
```

Our `package.json`

```json
{
  "name": "grg-frontend",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e",
    "i18n": "ng-xi18n --i18nFormat=xlf --output-path src/locale/"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^4.1.0",
    "@angular/common": "^4.1.0",
    "@angular/compiler": "^4.1.0",
    "@angular/core": "^4.1.0",
    "@angular/forms": "^4.1.0",
    "@angular/http": "^4.1.0",
    "@angular/platform-browser": "^4.1.0",
    "@angular/platform-browser-dynamic": "^4.1.0",
    "@angular/platform-server": "^4.1.0",
    "@angular/router": "^4.1.0",
    "core-js": "^2.4.1",
    "font-awesome": "^4.7.0",
    "markdown-to-html-pipe": "^1.2.1",
    "ng2-semantic-ui": "^0.7.0",
    "rxjs": "^5.1.0",
    "simplemde": "^1.11.2",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular/cli": "1.0.0",
    "@angular/compiler-cli": "^4.1.0",
    "@types/jasmine": "2.5.38",
    "@types/node": "~6.0.60",
    "codelyzer": "~2.0.0",
    "jasmine-core": "~2.5.2",
    "jasmine-spec-reporter": "~3.2.0",
    "karma": "~1.4.1",
    "karma-chrome-launcher": "~2.0.0",
    "karma-cli": "~1.0.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "karma-coverage-istanbul-reporter": "^0.2.0",
    "protractor": "~5.1.0",
    "ts-node": "~2.0.0",
    "tslint": "~4.5.0",
    "typescript": "~2.2.2"
  }
}
```

**Comment by edcarroll** [comment]
I see the issue - will get a fix out for this tomorrow.

**Comment by edcarroll** [comment]
Sorry for the delay - an immediate fix is to update typescript to `^2.3.0`, as the modals depend on a new feature of typescript (default generic parameters).

When I release `0.8.1` soon I'll include `typescript ^2.3.0` as a peer dependency, which will warn people that are using an older version of typescript.

**Comment by edcarroll** [comment]
`0.8.1` now includes `typescript^2.3.0` as a peer dependency, so people shouldn't run into this again.
